components/clarity-repl/src/clarity/analysis/types.rs

```diff
@@ -151,10 +151,23 @@ impl ContractAnalysis {
         self.read_only_function_types.get(name)
     }
 
-    pub fn get_private_function(&self, name: &str) -> Option<&FunctionType> {
+    pub fn get_private_function_type(&self, name: &str) -> Option<&FunctionType> {
         self.private_function_types.get(name)
     }
 
+    pub fn get_function_type(&self, name: &str) -> Option<(&str, &FunctionType)> {
+        if let Some(function_type) = self.public_function_types.get(name) {
+            return Some(("public".into(), function_type));
+        }
+        if let Some(function_type) = self.read_only_function_types.get(name) {
+            return Some(("read_only".into(), function_type));
+        }
+        if let Some(function_type) = self.private_function_types.get(name) {
+            return Some(("private".into(), function_type));
+        }
+        None
+    }
+
```

Comment:
Could we move this code in the `lsp` module instead? 
We're about to break the ability to compile `clarity-repl` to wasm, so we might have to code freeze that module.

Reply:
Sure thing, I'll try to make a pass on this PR today. It's been a bit abandoned